alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/processor/notification/NotificationProcessor.java

```diff
@@ -152,13 +225,32 @@ public NotificationProcessor(ConfigurationAccessor configurationAccessor, JobAcc
         return List.of();
     }
 
-    private Optional<ConfigurationModel> retrieveProviderConfig(ConfigurationJobModel job) {
+    private Optional<ConfigurationModel> retrieveProviderConfig(Long providerConfigID) {
         try {
-            return configurationAccessor.getConfigurationById(job.getProviderConfigIdAsLong());
+            return configurationAccessor.getConfigurationById(providerConfigID);
         } catch (AlertDatabaseConstraintException e) {
-            logger.error("Could not retrieve the provider config for job: {}", job.getName(), e);
+            logger.error("Could not retrieve the provider config for ID: {}", providerConfigID);
+            logger.debug(e.getMessage(), e);
             return Optional.empty();
         }
     }
 
+    private SetMap<NotificationFilterModel, AlertNotificationModel> extractNotificationInformation(List<AlertNotificationModel> notifications) {
+        SetMap<NotificationFilterModel, AlertNotificationModel> notificationFilterMap = SetMap.createDefault();
+        for (AlertNotificationModel alertNotificationModel : notifications) {
```

Comment:
Would we prefer streams over loops? We could convert this to a stream, map, then collect.

Reply:
We could use streams streams aren't always as performant as regular for loops.  But if there are optimizations for streams in subsequent releases of Java the code won't automatically get those performance benefits.  I would say it depends on running the performance tests that were written before changing over to streams.

Reply:
I would be curious how you would use a Stream here? Because we use the `AlertNotificationModel` to create a `NotificationFilterModel` and then put them both in the SetMap. Unless you are just suggesting I turn this for loop into ``` java
notifications.stream()
            .forEach( alertNotificationModel -> {
                NotificationFilterModel notificationFilterModel = extractNotificationInformation(alertNotificationModel);
                notificationFilterMap.add(notificationFilterModel, alertNotificationModel);
            });
```
Is this preffered?

Reply:
Oh you're right, it adds it to a completely separate object. Might as well leave it for now.